meta-mbl-distro/classes/mbl-partitions.bbclass

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) 2018 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+
+inherit mbl-artifact-names
+
+# ------------------------------------------------------------------------------
+# See docs/partitions.md for information about partition layouts. If you
+# change the names or values of these variables then make sure they are still
+# compatible with the .wks files in the wic directory and
+# recipes-core/base-files/files/fstab.
+# ------------------------------------------------------------------------------
+
+# Offsets of paritions should generally be flash erase block boundaries so that
+# writing to one partition is less likely to affect another partition, so we
+# have a variable for the flash erase block size from which to derive partition
+# offsets. The default value, 16MiB (16384KiB), is a fairly conservative erase
+# block size estimate. A more typical size for eMMC and SD Cards is 4MiB.
+#
+# Override this in a MACHINE's config when you have more accurate information
+# for that MACHINE.
+#
+# Override this in local.conf if your MACHINE uses removable storage media
+# (e.g. SD Cards) and you know the erase block sizes of the removal media that
+# will be used in your case.
+MBL_FLASH_ERASE_BLOCK_SIZE_KiB ?= "16384"
+
+
+# We have two different flavours of "partition" in MBL:
+# * Normal file system partitions, recorded in the partition table.
+# * Raw non-file system "partitions", not recorded in the partition table.
+
+# ------------------------------------------------------------------------------
+# Normal file system partitions, recorded in the partition table
+# ------------------------------------------------------------------------------
+MBL_PARTITIONS = "\
```

Comment:
I disagree, to a certain extent. If somebody changed `MBL_PARTITIONS` as things are right now, things could get messy because there's a bunch of other code that makes assumptions about the partitions.

I deliberately didn't use `?=` for the `MBL_*_DIR` variables because we have code that relies on the values, and it's not obvious where that code is. It's not straightforward for somebody to just change which directory something lives in.

In principle, I don't object to making it easy for a user to modify the partition table however they like, but if we do that, then we'll have to make sure that things won't break in non-obvious ways.

Reply:
I somewhat agree with you about the potential issues.